File: bench/perf/js/primes.js
function isPrime(n) {
    if (n < 2) return 0;
    if (n === 2) return 1;
    if (n % 2 === 0) return 0;
    let i = 3;
    while (i*i <= n) {
        if (n % i === 0) return 0;
        i += 2;
    }
    return 1;
}

let c = 0;
for (let k = 2; k <= 500000; k++) c += isPrime(k);
console.log(c);

File: bench/perf/rust/primes.rs
fn is_prime(n: i64) -> i64 {
    if n < 2 { return 0; }
    if n == 2 { return 1; }
    if n % 2 == 0 { return 0; }
    let mut i: i64 = 3;
    while i*i <= n {
        if n % i == 0 { return 0; }
        i += 2;
    }
    1
}

fn main() {
    let mut c: i64 = 0;
    for k in 2..=500000i64 {
        c += is_prime(k);
    }
    println!("{}", c);
}
